src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -128,8 +128,17 @@ CoordinatorInsertSelectExecScan(CustomScanState *node)
 
 			if (prunedTaskList != NIL)
 			{
-				ExecuteMultipleTasks(scanState, prunedTaskList, isModificationQuery,
-									 hasReturning);
+				TupleDesc tupleDescriptor = ScanStateGetTupleDescriptor(scanState);
+				bool randomAccess = true;
+				bool interTransactions = false;
+
+				Assert(scanState->tuplestorestate == NULL);
+				scanState->tuplestorestate =
+					tuplestore_begin_heap(randomAccess, interTransactions, work_mem);
+
+				ExecuteTaskListExtended(CMD_INSERT, prunedTaskList,
+										tupleDescriptor, scanState->tuplestorestate,
```

Comment:
@pykello Could you please make sure that we sort the tuplestore as we do in other executors? After rebasing my changes, I get some insert/select queries to have indeterministic returnings. Thanks!

Reply:
Sure, https://github.com/citusdata/citus/pull/2754